master/cdev.c

```diff
@@ -258,18 +259,18 @@ int eccdev_mmap(
  * \return Zero on success, otherwise a negative error code.
  */
 static int eccdev_vma_fault(
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0))
+#if (LINUX_VERSION_CODE < VM_FAULT_CHANGE_VERSION)
         struct vm_area_struct *vma, /**< Virtual memory area. */
 #endif
         struct vm_fault *vmf /**< Fault data. */
         )
 {
-    unsigned long offset = vmf->pgoff << PAGE_SHIFT;
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0))
+#if (LINUX_VERSION_CODE < VM_FAULT_CHANGE_VERSION)
     ec_cdev_priv_t *priv = (ec_cdev_priv_t *) vma->vm_private_data;
 #else
     ec_cdev_priv_t *priv = (ec_cdev_priv_t *) vmf->vma->vm_private_data;
 #endif
+    unsigned long offset = vmf->pgoff << PAGE_SHIFT;
```

Comment:
I think it is better to put this two lines up, because up to VM_FAULT_CHANGE_VERSION  `vma` instead of `vmf` was used.

Reply:
my understanding was that vma is only here if kernel < 4.10 but vmf always exists

Reply:
Is connected to the above resolved conversation.